### ISSUE TITLE
Docs: Explicitly say what the TS version warning error is in parser docs

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#7100]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Right now the docs on `warnOnUnsupportedTypeScript` don't actually show the complaint. It'd probably be good for SEO / general docs clarity to include it in the section:

```plaintext
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.1.0

YOUR TYPESCRIPT VERSION: 5.1.3

Please only submit bug reports when using the officially supported version.

=============
```

Accepting PRs to add a mention of this error message in the docs. ✨ 

### Affected URL(s)

https://typescript-eslint.io/architecture/parser#warnonunsupportedtypescriptversion

**Comment by akj2018** [comment]
Hi @JoshuaKGoldberg , I can definitely help you with this issue. Please assign it to me. 
Thanks.

Note : I am trying to generate a Note version of this line Parser.mdx
![image](https://github.com/typescript-eslint/typescript-eslint/assets/43956935/b225ec53-e91b-41bc-a970-a36379cf708c)

However, it's not getting rendered in Preview
![image](https://github.com/typescript-eslint/typescript-eslint/assets/43956935/23f2c18f-f32d-4c2a-af5f-64b278600a1f)

Can you help me with that ?

**Comment by JoshuaKGoldberg** [comment]
> Please assign it to me.

We can't assign to external folks (it's a limitation of GitHub), but feel free to work on this!

> it's not getting rendered in Preview

You'll need to develop the website locally per our [Local Development](https://typescript-eslint.io/contributing/local-development) guide. See https://typescript-eslint.io/contributing/local-development#website-development in particular. LMK if there's anything unclear we should add to the docs!

**Comment by Josh-Cena** [comment]
Also, you probably need to add line breaks:

```md
:::tip
Whatever
:::
```

**Comment by cojoclaudiu** [comment]
I got same issue, also I've noticed my liniting time doubled
